### PR TITLE
Completed first version of working RNAmigos endpoint

### PR DESCRIPTION
**Pull request by 1jayyaj1**
Tested with the json output from BP2. Output format might need to be updated once we know front-end team requirements.

### CODE REVIEW

**Review by ajaypatel24 — APPROVED** [review]

**Comment by rayanosseiran** [comment]
LGTM, but @larakollokian , thoughts on ditching route.py? would it be better to keep it for consistency across projects? 

p.s. add some java doc style [comments](https://stackoverflow.com/a/5339352)

**Review by rayanosseiran — CHANGES_REQUESTED** [review]
For consistency, could you return an error instead of blank output and raising exceptions? 

I'm a bit confused regarding the best ways to handle this though in the context of the pipeline. I don't think the way we've set-up error handling is very good (in my repos too).

+ Also, I don't think the graphs need to be treated as file input. They can be passed via form data since its just a dictionary (in the context of pipeline, you can/should make a separate endpoint with PDB)
+ Forgot to close temp file so it never gets destroyed until garbage collected

**Comment by larakollokian** [comment]
> thoughts on ditching route.py? would it be better to keep it for consistency across projects

Ya I agree, i think we should keep `routes.py` to keep it consistent with the other stuff. It also adds a layer of separation from the rest of the logic and looks cleaner imo

File: app.py [hunk 1]
```diff
@@ -1,13 +1,138 @@
...
-#app.register_blueprint(routes)
+
+
+@app.errorhandler(400)
+def resource_not_found(e):
+    return jsonify(error=str(e)), 400
+
+
+@app.route("/rnamigos", methods=['POST'])
+def rnamigos():
```
Comment (larakollokian) [inline app.py]: like @rayanosseiran mentionned, the routing should be done in `/routes/routes.py` to keep it consistent with BP2 and VeRNAl. It keeps all the layers separate and it just looks cleaner

File: app.py [hunk 2]
```diff
@@ -1,13 +1,138 @@
...
+
+    except:
+        print(traceback.format_exc())
+
+    # print(result_rnamigos)
+    return result_rnamigos
+
+
+@app.route("/rnamigos_pipeline", methods=['POST'])
+def rnamigos():
```
Comment (larakollokian) [inline app.py]: same as below v

File: rnamigos_launcher.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import pickle
+import json
+
+import networkx as nx
+
+from nx2fp import *
+from fp2lig import *
+
+
+def launch(G, library_path, n_hits=30):
```
Comment (larakollokian) [inline rnamigos_launcher.py]: maybe add more comments here, because it's a bit unclear on what this does at a first glance

**Review by larakollokian — CHANGES_REQUESTED** [review]
Overall this looks good, but I'm not really a fan of the structure of everything. 

1. I'm not sure why you got rid of the routing folder/file, but I think it would look cleaner if you put it back and put the routes there.
2. there are a lot of files at the root level that are related to the "business logic" of migos, and I think they should all be in a `core/` or `src/` folder instead. I'm talking about `fp2lig.py`, `nx2fp.py`, etc. You should have all the migos logic in a core/src folder, the routing in a `routes/` folder, and the root should have like config stuff, dockerfile and `app.py`. And app.py should be mostly empty, to keep it consistent with the other applications as well.
3. This isn't necessarily something that you should address in this PR, but I think it would be nice if we had more intuitive filenames and better comments for some of the modules. "fp2lig" doesn't really mean anything unless someone took the time to understand the code in detail. Maybe we should discuss this with Carlos and see what he thinks? Lemme know your thoughts.

**Pull request merged**